data/avro-schema/src/main/resources/avro/net/corda/data/virtualnode/VirtualNodeManagementResponse.avsc

```diff
@@ -0,0 +1,24 @@
+{
+  "type": "record",
+  "name": "VirtualNodeManagementResponse",
+  "doc": "The response to an {@link VirtualNodeRequest}.  On success the {@link VirtualNodeResponse#responseType} is {@link VirtualNodeManagementResponseSuccess}, on failure it is {@link VirtualNodeManagementResponseFailure}",
+  "namespace": "net.corda.data.virtualnode",
+  "fields": [
+    {
+      "name": "timestamp",
+      "type": {
+        "type": "long",
+        "logicalType": "timestamp-millis"
+      },
+      "doc": "Time ({@link java.time.Instant}) in milliseconds when the record was created."
+    },
+    {
+      "name": "responseType",
+      "doc": "Result of request, either a success or failure message",
+      "type": [
+        "net.corda.data.virtualnode.VirtualNodeManagementResponseSuccess",
+        "net.corda.data.virtualnode.VirtualNodeManagementResponseFailure"
```

Comment:
I think we can remove the extra wrapper `VirtualNodeManagementResponseSuccess`, and our management response just contain either a failure or the operation's response object, i.e. change this `responseType` to:

```
  {
      "name": "response",
      "type": [
        "net.corda.data.virtualnode.VirtualNodeCreateResponse",
        "net.corda.data.virtualnode.VirtualNodeManagementResponseFailure"
       ]
```
WDYT?

The RPCSender code can just check if the type of response is a failure then handle failure, otherwise it was a success. Might just simplify the model.

Reply:
I could do that - but that means you wouldn't have the wrapper in place for when update becomes a thing.

IMO I like the wrapper for the sake of readability as on the sending end of the request - you can parse the response as 
```kotlin
when (result) {
  success -> ..
  fail -> ..
}
```

Reply:
:D I talked myself around to your idea I see its the pattern used in Entity stuff, so I don't mind either way.